packages/common/src/hooks/useListData/useQueryParams.ts

```diff
@@ -0,0 +1,27 @@
+import { useEffect } from 'react';
+import { useRowRenderCount } from './useRowRenderCount';
+import { usePagination, PaginationState } from './usePagination';
+import { useSortBy, SortState } from './useSortBy';
+
+interface QueryParams<T> extends SortState<T>, PaginationState {}
+
+interface QueryParamsState<T> extends SortState<T>, PaginationState {
+  pagination: PaginationState;
+  queryParams: QueryParams<T>;
+}
+
+export const useQueryParams = <T>(
+  initialSortBy: keyof T
+): QueryParamsState<T> => {
+  const numberOfRows = useRowRenderCount();
+  const pagination = usePagination(numberOfRows);
+  const { sortBy, onChangeSortBy } = useSortBy(initialSortBy);
+
+  useEffect(() => {
+    pagination.onChangeFirst(numberOfRows);
+  }, [numberOfRows]);
+
+  const queryParams = { ...pagination, pagination, sortBy, onChangeSortBy };
+
+  return { ...pagination, pagination, sortBy, onChangeSortBy, queryParams };
```

Comment:
bit of a weird return.. I just really hate it when I can't destructure one value but also keep the whole thing as a chunk..!